bitcoin/src/string.rs

```diff
@@ -35,6 +35,9 @@ pub trait FromHexStr: Sized {
 }
 
 /// Hex parsing error
+///
+/// Does not implement [`internals::impl_from_infallible`] because there is already a blanket `impl
+/// From<E> for FromHexError<E>` defined which converts the error into `FromHexError::ParseHex(E)`.
```

Comment:
I think it's fine. I believe conversion from infallible is more rare than conversion from inner error type.